ktor-shared/ktor-serialization/ktor-serialization-kotlinx/common/src/io/ktor/serialization/kotlinx/KotlinxSerializationConverter.kt

```diff
@@ -72,7 +72,7 @@ public class KotlinxSerializationConverter(
         val fromExtension = extensions.asFlow()
             .map { it.deserialize(charset, typeInfo, content) }
             .firstOrNull { it != null || content.isClosedForRead }
-        if (fromExtension != null || content.isClosedForRead) return fromExtension
+        if (extensions.isNotEmpty() && (fromExtension != null || content.isClosedForRead)) return fromExtension
```

Comment:
Should we also add this check in `serializeNullable(..) { ... if (fromExtension != null) return fromExtension ... }`?

Reply:
no, because if `fromExtension != null` then `extensions` are always not empty